### PR TITLE
Add E2E tests for Kubernetes 1.20/1.21

### PR DESCRIPTION
**Pull request by dmitsh**
# Description

Add E2E tests for Kubernetes 1.20/1.21

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #3366 #3348

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by wcs1only — DISMISSED** [review]
LGTM, so long as the new tests pass.

**Comment by wcs1only** [comment]
One note: if we hit any kind of rate limiting on KinD builds, we probably don't need HA-mode + non-HA-mode for all 3/5 versions. But, we can reduce the number of non-HA-mode KinD runs at a future time, only if necessary.

**Comment by daixiang0** [comment]
lgtm

btw, do we have a doc comment lowest k8s support version?

**Comment by dmitsh** [comment]
> lgtm
> 
> btw, do we have a doc comment lowest k8s support version?

We are discussing it here: https://github.com/dapr/dapr/issues/3348

**Comment by yaron2** [comment]
Can we remove 1.17? I want to align to supported k8s versions.

File: .github/workflows/dapr-e2e-kind.yml [hunk 1]
```diff
@@ -43,29 +43,37 @@ jobs:
       fail-fast: false # Keep running if one leg fails.
       matrix:
         k8s-version:
-        - v1.17.11
-        - v1.18.8
-        - v1.19.1
+        - v1.17.17
```
Comment (yaron2) [inline .github/workflows/dapr-e2e-kind.yml:46]: Remove

File: .github/workflows/dapr-e2e-kind.yml [hunk 2]
```diff
@@ -43,29 +43,37 @@ jobs:
...
+        - v1.18.19
```
Comment (yaron2) [inline .github/workflows/dapr-e2e-kind.yml:47]: Remove

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Review by yaron2 — APPROVED** [review]
lgtm

**Review by yaron2 — APPROVED** [review]
lgtm

**Pull request merged**